test/uk/gov/hmrc/transitmovementsrouter/services/ObjectStoreConnectorSpec.scala

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.transitmovementsrouter.services
+
+import org.scalatest.matchers.must.Matchers
+import org.scalatest.BeforeAndAfterEach
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.freespec.AnyFreeSpec
+import org.scalatest.time.Seconds
+import org.scalatest.time.Span
+import org.scalatestplus.mockito.MockitoSugar
+import org.scalatestplus.scalacheck.ScalaCheckPropertyChecks.forAll
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.objectstore.client.RetentionPeriod.SevenYears
+import uk.gov.hmrc.objectstore.client.config.ObjectStoreClientConfig
+import uk.gov.hmrc.transitmovementsrouter.base.TestActorSystem
+import uk.gov.hmrc.transitmovementsrouter.fakes.objectstore.ObjectStoreStub
+import uk.gov.hmrc.transitmovementsrouter.generators.TestModelGenerators
+import uk.gov.hmrc.transitmovementsrouter.models.responses.UpscanResponse.DownloadUrl
+
+import java.time.Clock
+import java.util.UUID.randomUUID
+
+class ObjectStoreServiceSpec
+    extends AnyFreeSpec
+    with Matchers
+    with MockitoSugar
+    with ScalaFutures
+    with TestModelGenerators
+    with TestActorSystem
+    with BeforeAndAfterEach {
+
+  val baseUrl = s"http://baseUrl-${randomUUID().toString}"
+  val owner   = s"owner-${randomUUID().toString}"
+  val token   = s"token-${randomUUID().toString}"
+  val config  = ObjectStoreClientConfig(baseUrl, owner, token, SevenYears)
+
+  implicit val hc = HeaderCarrier()
+  implicit val ec = materializer.executionContext
+
+  lazy val objectStoreStub = new ObjectStoreStub(config)
```

Comment:
@drnaylor-oc with service generally we mock connector or client. but here you are creating stub for service testing. I think this should be done by mocking ```  PlayObjectStoreClientEither ``` ? Also I can not see test cases for if something got failed with Object store while uploading the file to object store. The failed one test cases written with this class is looks like for if download url is incorrect.

Reply:
> I think this should be done by mocking PlayObjectStoreClientEither ?

Nope. This is fine. We do use fakes, and I prefer them when we're not changing their behaviour.

> Also I can not see test cases for if something got failed with Object store while uploading the file to object store. The failed one test cases written with this class is looks like for if download url is incorrect. we should also include one test case for if Object store return failed Future.

Yes, we need a test for that.

Reply:
@olakunlekuye2 please add the test case for object store failed future

Reply:
Okay I'll do that